docs/docs_updated_test.go

```diff
@@ -0,0 +1,46 @@
+package docs
+
+import (
+	"flag"
+	"strings"
+	"testing"
+
+	"github.com/grafana/agent/component"
+	_ "github.com/grafana/agent/component/all"
+	"github.com/grafana/agent/docs/generator"
+	"github.com/stretchr/testify/require"
+)
+
+// Run the below generate command to automatically update the Markdown docs with generated content
+//go:generate go test -run TestCompatibleComponentsSectionUpdated -fix-tests
+
+var fixTestsFlag = flag.Bool("fix-tests", false, "update the test files with the current generated content")
+
+func TestCompatibleComponentsSectionUpdated(t *testing.T) {
```

Comment:
If someone forgets to update the docs, CI will fail. But updating them is also as easy as running `go generate` for this file.